File: app_mobile/lib/models/devices/light.dart
import 'package:flutter/material.dart';

import '../device.dart';

class LightData {
  static DeviceRequestDevice getDeviceRequest({@required String uid}) {
    return DeviceRequestDevice(
      uid: uid,
      action: "toggle",
      gpio: 1,
    );
  }
}

File: types/src/light.ts
import { Client } from '.';

export namespace Light {
  export const createLightRequestByUID = ({
    uid,
  }: {
    uid: string;
  }): Client.DeviceRequestByUID['device'] => {
    return {
      uid,
      action: 1,
    };
  };
}
